torch/distributed/tensor/_sharding_prop.py

```diff
@@ -366,6 +367,41 @@ def propagate_op_sharding_non_cached(self, op_schema: OpSchema) -> OutputShardin
         # special case op, we don't need to propagate for local
         # scalar. TODO: figure out a better way to handle this
         if op_schema.op is aten._local_scalar_dense.default:
+            # there should only be one argument for _local_scalar_dense
+            arg = op_schema.args_schema[0]
+
+            if isinstance(arg, DTensorSpec):
+                has_partial = False
+                for placement in arg.placements:
+                    if isinstance(placement, Partial):
+                        has_partial = True
+                        break
+
+                if has_partial:
```

Comment:
something like `if any(p.is_partial() for p in arg.placements)` should work instead of the block above?